server/channels/app/imaging/orientation.go

```diff
@@ -55,22 +56,77 @@ func MakeImageUpright(img image.Image, orientation int) image.Image {
 	}
 }
 
-// GetImageOrientation reads the input data and returns the EXIF encoded
-// image orientation.
-func GetImageOrientation(input io.Reader) (int, error) {
-	exifData, err := exif.Decode(input)
-	if err != nil {
-		return Upright, fmt.Errorf("failed to decode exif data: %w", err)
+type fwSeeker struct {
+	r io.Reader
+}
+
+func (f *fwSeeker) Read(p []byte) (n int, err error) {
+	return f.r.Read(p)
+}
+
+func (f *fwSeeker) Seek(offset int64, whence int) (int64, error) {
+	// We don't support seeking backwards.
+	if whence != io.SeekCurrent {
+		return 0, fmt.Errorf("unsupported whence value: %d", whence)
 	}
 
-	tag, err := exifData.Get("Orientation")
+	// Seeking forward means we can simply discard the data.
+	n, err := io.CopyN(io.Discard, f.r, offset)
 	if err != nil {
-		return Upright, fmt.Errorf("failed to get orientation field from exif data: %w", err)
+		return n, fmt.Errorf("failed to seek: %w", err)
 	}
 
-	orientation, err := tag.Int(0)
-	if err != nil {
-		return Upright, fmt.Errorf("failed to get value from exif tag: %w", err)
+	return n, nil
```

Comment:
Seek should return the current absolute position within the stream. Right now this is returning the number of bytes read, which is usually different. We should track the position in a separate field in the struct and return that.

Reply:
Good point. I was making a bit too many assumptions on the underlying implementation.